now/run_backend.py

```
@@ -205,3 +206,161 @@ def estimate_request_size(index, max_request_size):
     max_size = 50_000
     request_size = max(min(max_request_size, int(max_size / size)), 1)
     return request_size
+
+
+def update_dict_with_no_overwrite(dict1: Dict, dict2: Dict):
+    """
+    Update dict1 with dict2, but only if the key does not exist in dict1
+
+    :param dict1: dict to be updated
+    :param dict2: dict to be used for updating
+    """
+    for key, value in dict2.items():
+        if key not in dict1:
+            dict1[key] = value
+
+
+def create_dataclass(user_input: UserInput):
+    """
+    Create a dataclass from the user input using the selected search anf filter fields
+    and their corresponding modalities
+
+    :param user_input: user input
+    """
+    all_annotations = {}
+    all_class_attributes = {}
+    all_modalities = {}
+    all_modalities.update(user_input.search_fields_modalities)
+    update_dict_with_no_overwrite(all_modalities, user_input.filter_fields_modalities)
```

I was wondering if instead of ignoring it, we should raise an error in case there's an overlap between search and filter fields..

---

it is expected to be an overlap, in fact these dictionaries contain candidates for search and filter fields so some fields could be used for search or for filtering. We create these dicts before the user chooses therefore there's an overlap